modules/ebi-metagenomics/mapseq2biom/main.nf

```diff
@@ -0,0 +1,59 @@
+
+process MAPSEQ2BIOM {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:0.1.1--pyhdfd78af_0' :
+        'biocontainers/mgnify-pipelines-toolkit:0.1.1--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(msq)
+    tuple path(db_fasta), path(db_tax), path(db_otu), path(db_mscluster), val(db_label)
```

Comment:
Just an idea on how to make the second tuple more tidy.  Instead of passing that many arguments, you could have something like these:

`tuple val(meta2), path(db_files)`

Where meta2.id = "SILVA-SSU"
And in db_files you have the list of db files. As all of them have the same prefix and different extensions, you can then find the db_otu file like this:

DB_OTU =`find -L ./ -name "*.otu" `

And then do:
```
   mapseq2biom \
        $args \
        --krona ${prefix}.txt \
        --no-tax-id-file ${prefix}.notaxid.tsv \
        --label ${meta2.id} \ # This can be define as a second prefix at the beginning pf the script block code
        --query ${msq} \
        --otu-table \$DB_OTU \
        --out-file ${prefix}.tsv

```

Reply:
@vagkaratzas yes the db_label always has to be given 

@Ales-ibt this is a really nice idea thank you!! however the values these inputs could have is very simple, they're always going to be 'hardcoded' in a pipeline that calls this module, so I don't think adding meta provides much functionality. That said, your comment made me realise that I definitely have extra inputs in there that are not even used (db_fasta and db_tax) so I will remove them. Thank you! 😄